test/externalTests/README.md

```diff
@@ -0,0 +1,66 @@
+    - The fewer commits in these branches, the better. Ideally, any changes needed to make the compiler
+        work should be submitted upstream and the branches should just be tracking the upstream
+        one without any extra commits.
+4. Create a script for compiling/testing the project and put it in `test/externalTests/` in the
+    Solidity repository.
+    - The script should apply workarounds necessary to make the project actually use the compiler
+      binary it receives as a parameter and possibly apply some generic workarounds that should
+      work across different versions of the upstream project. Very specific workarounds that may
+      easily break with every upstream change are better done as commits in the fork.
+5. Add the script to `tests/externalTests.sh`.
+6. Add the test to CircleCI configuration. Make sure to add both a compilation-only run and one that
+    also executes the test suite. If the latter takes a significant amount of time (say, more
+    than 15 minutes) make it run nightly rather than on every PR.
+
+#### Pulling upstream changes into a fork
+1. Pull changes directly into the main branch in the fork. This should be straightforward thanks to
+    it not containing any of our customizations.
+2. If the update is straightforward and looks safe, go straight to point 5. Otherwise you need to
+    test it first.
+3. Create new branches corresponding to the two versioned ones but suffixed with `_new`. E.g.
+    `master_070_new` and `master_080_new`. Then rebase them so that they are on top of the updated
+    main branch. This may require tweaking some of the commits to apply our fixes in new places.
+4. Create PRs on `deveop` and `breaking` in Solidity repo which modify the script to use the new
+    branches instead of `master_070`/`master_080`. Tweak the new branches until external tests
+    in CI pass in the PRs.
+5. Discard the PRs and move the original branches to the place where the `_new` ones are and remove
+    the `_new` branches.
+
+#### Changes needed after a breaking release of the compiler
+When a breaking version of the compiler gets released and becomes the most recent release, the scripts
+and the branches in the forks need to be updated:
+- In each fork create a branch corresponding to the new brekaing version. E.g. if the current
+    branches are `master_070` and `master_080` the new one should be called `master_090`.
+- Leave the oldest (i.e. `master_070`) branch as is. We will not be updating it any more but there is no
+    need to remove it.
+- Update scripts on `develop` to now refer to the branch that used to be breaking (i.e. `master_080`)
+    and on `breaking` to refer to the newly added branch (i.e. `master_090`).
+    - Take care not to overwrite these changes when merging `develop` into breaking. Each branch
+        should always use the branches from the external repos that correspond to it.
```

Comment:
Maybe this should be a part of the release checklist. Do we have one for breaking releases?

Reply:
I think in practice we'll only do it the first time an external test breaks, so since https://github.com/ethereum/solidity/pull/10556/files#r540011902 might be true, I don't think we need to put it in the release checklist.

Reply:
This is rewritten now. The new text suggests that it's fine to keep using the older branch and does not require doing anything on a breaking release..